x/profiles/keeper/keeper_test.go

```diff
@@ -216,3 +215,162 @@ func (suite *KeeperTestSuite) TestKeeper_GetProfiles() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestKeeper_SaveDTagTransferRequest() {
+	tests := []struct {
+		name                  string
+		storedTransferReqs    []types.DTagTransferRequest
+		transferReq           types.DTagTransferRequest
+		expErr                error
+		expStoredTransferReqs []types.DTagTransferRequest
+	}{
+		{
+			name: "already present request returns error",
+			storedTransferReqs: []types.DTagTransferRequest{types.NewDTagTransferRequest(
+				suite.testData.user, suite.testData.otherUser),
+			},
+			transferReq: types.NewDTagTransferRequest(suite.testData.user, suite.testData.otherUser),
+			expErr: fmt.Errorf("the transfer request from %s to %s has already been made",
+				suite.testData.otherUser, suite.testData.user),
+			expStoredTransferReqs: []types.DTagTransferRequest{types.NewDTagTransferRequest(
+				suite.testData.user, suite.testData.otherUser)},
+		},
+		{
+			name:               "not already present request saved correctly",
+			storedTransferReqs: nil,
+			transferReq:        types.NewDTagTransferRequest(suite.testData.user, suite.testData.otherUser),
+			expErr:             nil,
+			expStoredTransferReqs: []types.DTagTransferRequest{types.NewDTagTransferRequest(
+				suite.testData.user, suite.testData.otherUser)},
+		},
```

Comment:
Can you add some tests with an already existing request? Particularly: 

- storing a new request with a different current owner address works 
- storing a new request with a different receiver address works
- storing a new request with a different DTag works